extensions/spring-xd-extension-gpload/src/main/java/org/springframework/xd/gpload/GploadEnvironmentProvider.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.xd.gpload;
+
+import org.springframework.batch.step.tasklet.x.EnvironmentProvider;
+
+import java.util.Map;
+
+/**
+ * Class that sets environment variables for gpload
+ */
+public class GploadEnvironmentProvider implements EnvironmentProvider {
+
+	String gploadHome;
+
+	boolean usePasswordFile;
+
+	String passwordInfo;
+
+	public GploadEnvironmentProvider(String gploadHome, boolean usePasswordFile, String passwordInfo) {
+		this.gploadHome = gploadHome;
+		this.usePasswordFile = usePasswordFile;
+		this.passwordInfo = passwordInfo;
+	}
+
+	@Override
+	public void setEnvironment(Map<String, String> env) {
+		env.put("GPHOME_LOADERS", gploadHome);
```

Comment:
Hmm, that's a problem with that startup script. I guess we could check for ${gploadHome}/greenplum_loaders_path.sh and if it is not there check for ${gploadHome}/greenplum_path.sh and set GPHOME in that case.